### PR TITLE
Updated stats

### PR DESCRIPTION
**Pull request by hthompson6**
Current stats implementation is out of date as of the release of ACOS 4.10. This update returns a valid compilation of (loadbalancer/virtual-server) statistics.

### CODE REVIEW

**Comment by hthompson6** [comment]
check

**Comment by hthompson6** [comment]
check

**Comment by hthompson6** [comment]
check

**Comment by hthompson6** [comment]
check

File: a10_neutron_lbaas/v2/handler_lb.py [hunk 1]
```diff
@@ -67,23 +67,27 @@ def delete(self, context, lb):
 
     def stats(self, context, lb):
         with a10.A10Context(self, context, lb) as c:
+            total_fwd_bytes, total_rev_bytes, curr_conns, tot_conns = 0, 0, 0, 0
             try:
                 name = self.meta(lb, 'id', lb.id)
                 r = c.client.slb.virtual_server.stats(name)
-
+                for port in r["virtual-server"]["port-list"]:
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/v2/handler_lb.py]: Did we have this incorrect or did we get bitten by upstream schema changes?

**Comment by mdurrant-b3** [comment]
:-1: 
```
acos_client retried https://10.48.7.107:443/axapi/v3/slb/virtual-server/a9a20672-be4f-4efd-9a3d-2fb1102fdd40/stats 0 times
2016-11-17 17:08:22.614 1860 DEBUG acos_client.v30.axapi_http [req-e4c2885a-ad99-4002-a5be-541afb938f4e admin -] acos_client retried https://10.48.7.107:443/axapi/v3/slb/virtual-server/a9a20672-be4f-4efd-9a3d-2fb1102fdd40/stats 0 times request /usr/local/src/acos-client/acos_client/v30/axapi_http.py:139
axapi_http: data = {
    "response": {
        "status": "fail",
        "err": {
            "msg": "No stats definition corresponding to the URI.",
            "code": 1023656965,
            "from": "JSON"
        }
    }
}
```

File: a10_neutron_lbaas/v2/handler_lb.py [hunk 2]
```diff
@@ -67,23 +67,27 @@ def delete(self, context, lb):
...
-                    "bytes_out": r["virtual_server_stat"]["resp_bytes"],
-                    "active_connections":
-                        r["virtual_server_stat"]["cur_conns"],
-                    "total_connections": r["virtual_server_stat"]["tot_conns"]
+                    "bytes_in": total_fwd_bytes,
+                    "bytes_out": total_rev_bytes,
+                    "active_connections": curr_conns,
+                    "total_connections": tot_conns
                 }
             except Exception:
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/v2/handler_lb.py]: We need to refine our exception catching here - this is eating driver errors.

Comment (mdurrant-b3) [inline a10_neutron_lbaas/v2/handler_lb.py]: This is why tests tempest tests passed with 404 errors happening in acos.

File: a10_neutron_lbaas/v2/handler_lb.py (hunk 1 above)
Comment (hthompson6) [inline a10_neutron_lbaas/v2/handler_lb.py]: Bitten by upstream schema changes

**Comment by hthompson6** [comment]
In regards to the log, the changes have been made to the acos client to correct for this and it is waiting to be merged.

File: a10_neutron_lbaas/v2/handler_lb.py (hunk 2 above)
Comment (hthompson6) [inline a10_neutron_lbaas/v2/handler_lb.py]: Yes catching all exceptions can be problematic. I propose catching only NoneType errors only. The only other exceptions that are likely to be raised are AcosErrors and KeyErrors. This, and perhaps additional/enhanced testing, will give us greater visibility of further issues caused by future upstream changes.

**Comment by hthompson6** [comment]
Being replaced by implementation of extended stats. All code in this PR will be overwritten by that PR.

**Pull request closed without merging**